### PR TITLE
Gates the Graph details E2E specs on state they establish themselves

### PR DESCRIPTION
**Pull request by nikolay-1986**
Two specs went red on every editor in the first E2E run since Aug 4 — [run 31586914489](https://github.com/gitkraken/vscode-gitlens/actions/runs/31586914489) on `main` @ `0b3b5776c`: 2 failed / 210 passed on VS Code, and the same two among the failures on Windsurf and Positron, 3 of 3 attempts each. Both were reproduced locally and bisected to a **layout change, not a product defect**. No product code is touched here.

### What actually broke

`43d7c5df5` ("Updates the GitLens views layout to reveal some of the Welcome view") gave `gitlens.views.welcome` `visibility: visible` with `initialSize: 3` against the Graph's 7, where it had shipped collapsed at 1. Measured live at the default side-bar width: Graph 299×544 with Welcome holding 299×220 under it. That height comes off the Graph, and its details panel takes the loss.

Neither spec was asserting quite what it looked like it was asserting:

**`treeView.test.ts:162`** — the details file tree is virtualized, so `lit-virtualizer` mounts rows only for its own viewport. With a commit selected the panel's header, message and AI blocks fill 235px on their own, leaving `gl-tree-view` at y=531 of a 544px viewport with **zero** `gl-tree-item` in the DOM — the same observation as a tree that failed to render. The data was never missing: the pane reads "Files changed 1", and scrolling the 42px to the end of the panel mounts the row. A/B on the layout commit alone: 5 of 5 passing at `43d7c5df5^`, `:162` failing at `43d7c5df5`. Width can't substitute — the panel only moves beside the graph past ~820px, so below that it keeps splitting the height.

**`graphConflictResolution.test.ts:295`** — `+hasConflicts` is carried by the WIP details header's overflow chip, so in this fixture it exists only while the WIP row is the selection (with a commit selected the panel renders `gl-details-commit-panel` and there is no `gl-details-wip-header` to carry it; measured both ways). The overview bar carries a selection-independent copy, but it never renders here — `gitlens.graph.overviewBar.visibility` defaults to `worktrees` and the fixture has one. The gate waited for that context without selecting anything, which held only because a fresh graph auto-selects the WIP row. It also has to leave resolve mode first: the mode panel renders in the WIP details' place and the webview is retained across `resetUI`, so a mode this file's `afterEach` failed to close surfaced here as "the WIP details never rendered".

### The change

Each gate now establishes what it measures. `scrollDetailsToFileTree` brings the tree into the panel's viewport (retried together with the row assertion, since `.details-content` is reused across selection changes and an early scroll gets clamped when taller content swaps in), and the conflict gate opens the details panel, exits resolve mode and selects the WIP row before polling for the context. The conflict describe raises its per-test timeout to 90s, because those stages each carry their own timeout and could otherwise add past the 60s cap — firing mid-stage and reporting a bare "Test timeout" naming none of them.

Scrolling rather than growing the side bar is deliberate, and was tried the other way first. Collapsing the Welcome pane works in isolation (Graph 544 → 764, tree at y=641 with its items mounted) but not as a harness primitive: pane sizes are shared mutable state across spec files in a worker, and a collapsed pane did not stay collapsed — one run had it collapsed for `graphConflictResolution` and expanded again for `treeView`. Worse, a real click on the pane header starts a **drag** (`draggable="true"`), which re-homes the pane and left the graph painting no rows at all for every later spec. A panel's own scroll position is local to the assertion that needs it.

### Validation

Locally against VS Code (`--workers=1 --retries=0`):

- `treeView` — 5 passed (16s; was failing at 31s)
- `graphConflictResolution` — 7 passed (24s)
- `treeView` + `graphConflictResolution` + `graphHeader` + `graphReview` + `smoke` + `graphRowActions` — **42 passed**

`oxlint --type-aware --type-check` and `tsc` clean on the changed files. An independent review of the first commit produced five findings, all addressed in the second (unretried scroll, gate ordering vs. `exitResolveMode`'s single probe, the timeout budget, a comment that overstated where the context lives, and the per-file `+conflict` context riding the same virtualized tree).

**This PR exists partly to get the fixes onto CI**: E2E runs on `workflow_dispatch` only, so the fork legs (Windsurf, Positron) can't be exercised locally here.

### Not addressed here

`graphDetails.test.ts:265` (`.details-content` stays visible after the *Hide Details Panel* toggle) fails on Windsurf and Positron in CI, and locally on VS Code too — including with these changes stashed, and its helper is byte-for-byte unchanged by them. Same area, different mechanism (the split panel keeps the pane mounted at `position: 100` with `inert`, so a residual 1–2px box may keep it "visible" to Playwright). Tracked separately.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/e2e/specs/graphConflictResolution.test.ts [hunk 1]
```diff
@@ -241,8 +267,18 @@ test.describe('Graph — Conflict Resolution', () => {
 		await selectWipDetails(webview);
 
 		// Both conflicted files render a `gitlens:file…+conflict…` context; assert on the one for
```
[the remark this answers is not in the document]
Comment (nikolay-1986) [reply tests/e2e/specs/graphConflictResolution.test.ts:270]: Good catch, and the header was worse off than this describes: it also contradicted itself ten lines down, where it lists `+conflict` among what the file covers.

The claim went stale when #5548 (conflicted WIP details file rows not exposing the context under the new graph engine) was fixed and the test came back — so it predates this PR, but this PR adds the scroll that makes that assertion reliable, so it belongs here. Fixed in a8dd7f4f4: the header now states that the per-file context is covered, that #5548 is fixed, and that it rides the details panel's virtualized file tree, which is why the spec scrolls before asserting.

File: tests/e2e/graphHelpers.ts [hunk 2]
```diff
@@ -38,6 +43,26 @@ export async function widenSideBarForGraph(vscode: VSCodeInstance, steps = 12):
...
+ * The tree is virtualized: `lit-virtualizer` mounts rows only for its own viewport, so a tree below the
+ * fold has NO `gl-tree-item` in the DOM at all — indistinguishable from a tree that failed to render.
+ * With a commit selected that is the default state on a side-bar-sized panel: the header, message and
+ * AI blocks fill the panel on their own. Measured at a ~300px-wide side bar with the Welcome pane
+ * expanded (its shipped `visibility`, holding ~220px of the height): `gl-tree-view` sits at y=531 of a
+ * 544px viewport with 0 items, and scrolling the 42px to the end of the panel mounts them.
+ *
+ * Scrolling rather than growing the side bar on purpose: pane sizes are shared mutable state across
+ * spec files in a worker, and a pane this collapses does not stay collapsed — the panel's own scroll
+ * position is local to the assertion that needs it.
```
[the remark this answers is not in the document]
Comment (nikolay-1986) [reply tests/e2e/graphHelpers.ts]: Agreed — reworded in a8dd7f4f4. The sentence is now split in two, keeping the reason it scrolls instead of freeing height: pane sizes are shared mutable state across spec files in a worker, so collapsing a pane does not keep it collapsed for the specs that follow, whereas a panel's own scroll position is local to the assertion that needs it.
